api/catalog_test.go

```diff
@@ -36,31 +35,35 @@ func TestCatalog_Nodes(t *testing.T) {
 
 	catalog := c.Catalog()
 
-	if err := testutil.WaitForResult(func() (bool, error) {
+	for r := retry.OneSec(); r.NextOr(func() { t.Fatal("no nodes") }); {
 		nodes, meta, err := catalog.Nodes(nil)
 		if err != nil {
-			return false, err
+			t.Log("catalog.Nodes: ", err)
+			continue
 		}
-
 		if meta.LastIndex == 0 {
-			return false, fmt.Errorf("Bad: %v", meta)
-		}
-
-		if len(nodes) == 0 {
-			return false, fmt.Errorf("Bad: %v", nodes)
-		}
-
-		if _, ok := nodes[0].TaggedAddresses["wan"]; !ok {
-			return false, fmt.Errorf("Bad: %v", nodes[0])
-		}
-
-		if nodes[0].Datacenter != "dc1" {
-			return false, fmt.Errorf("Bad datacenter: %v", nodes[0])
-		}
-
-		return true, nil
-	}); err != nil {
-		t.Fatal(err)
+			t.Log("got last index 0 want > 0")
+			continue
+		}
+		want := []*Node{
+			{
+				ID:         s.Config.NodeID,
+				Node:       s.Config.NodeName,
+				Address:    "127.0.0.1",
+				Datacenter: "dc1",
+				TaggedAddresses: map[string]string{
+					"lan": "127.0.0.1",
+					"wan": "127.0.0.1",
+				},
+				Meta:        map[string]string{},
+				CreateIndex: meta.LastIndex - 1,
```

Comment:
This could be a little nondeterministic, but we can see how it goes with the newer convergence timing.